tests/keras/preprocessing/sequence_test.py

```diff
@@ -165,13 +167,90 @@ def test_TimeseriesGenerator():
                                    length=10, sampling_rate=2,
                                    start_index=10, end_index=30,
                                    batch_size=2)
-
-    assert len(data_gen) == 5
+    assert len(data_gen) == 6
     assert np.allclose(data_gen[0][0], np.array(
         [np.array(data[10:19:2]), np.array(data[11:20:2])]))
     assert (np.allclose(data_gen[0][1],
                         np.array([targets[20], targets[21]])))
 
+    with assert_raises(ValueError) as context:
+        TimeseriesGenerator(data, targets, length=50)
+    error = str(context.exception)
+    assert '`start_index+length=50 > end_index=49` is disallowed' in error
+
+
+def test_TimeSeriesGenerator_doesnt_miss_any_sample():
+    x = np.array([[i] for i in range(10)])
+
+    for length in range(3, 10):
+        g = TimeseriesGenerator(x, x,
+                                length=length,
+                                batch_size=1)
+        expected = max(0, len(x) - length)
+        actual = len(g)
+
+        assert expected == actual, \
```

Comment:
You don't need this error message since it's a unit test. If you prefer including it, please format without the use of `\` to follow codebase style